File: src/kipper/effects/ShipTrails.py
from kipper.effects import ParticleEmitterConfig
from kipper.effects.transitions import Linear
from kipper.effects.transitions import EaseInQuad
from kipper.effects.transitions import EaseOutCubic
from kipper.effects.transitions import EaseOutQuad

import math
from random import random

L = Linear()
E = EaseOutQuad()

class SampleConfigImpl(ParticleEmitterConfig):

    def getMaxParticles(self):
        return 25

    def getDurationTicks(self):
        return 20

    def getSpawnRate(self):
        return 0.5

    def isContinuous(self):
        return True

    def isRectShape(self, p):
        return True

    def getSize(self, p):
        a = 12
        b = 1
        return int(L.call(p.ticks, a, b-a, self.getDurationTicks()))

    def getTheta(self, p):
        return math.pi

    def getSpeed(self, p):
        return 2

    def getHue(self, p):
        return 210.0/360.0

    def getSaturation(self, p):
        a = 10
        b = 100
        return E.call(p.ticks, a, b-a, self.getDurationTicks())/100

    def getBrightness(self, p):
        return 1

File: src/kipper/effects/ShipTrails.java
package kipper.effects;

import kipper.effects.transitions.*;

public class ShipTrails extends ParticleEmitterConfig
{
    @Override public int getMaxParticles() { return 25; }
    @Override public int getDurationTicks() { return 20; }
    @Override public double getSpawnRate() { return 0.5; }
    @Override public boolean isContinuous() { return true; }
    @Override public boolean isRectShape(Particle p) { return false; }
    @Override public int getSize(Particle p) { int a = 12; int b = 1; return (int)(new Linear().call(p.ticks, a, b-a, getDurationTicks())); }
    @Override public double getTheta(Particle p) { return Math.PI; }
    @Override public double getSpeed(Particle p) { return 2; }
    @Override public float getHue(Particle p) { return 210f / 360f; }
    @Override public float getSaturation(Particle p) { int a = 10; int b = 100; return (float)(new EaseOutQuad().call(p.ticks, a, b-a, getDurationTicks())/100); }
    @Override public float getBrightness(Particle p) { return 1; }
}
